### PR TITLE
Refactor chat-plugins and formats to for...of

### PR DESCRIPTION
**Pull request by KrisOnline**
In the instances of ``for (let x of y)``, I didn't use `const` because they were modified within the loop and I don't like declaring needless variables just do modify them if we use `const`. However, if `const` is supposed to be standard, I will do that.

### CODE REVIEW

File: chat-plugins/datasearch.js [hunk 1]
```diff
@@ -538,15 +536,15 @@ function runDexsearch(target, cmd, canAll, message) {
 			if (target === 'zrecovery') {
 				if (parameters.length > 1) return {reply: "The parameter 'zrecovery' cannot have alternative parameters"};
 				let recoveryMoves = ["aromatherapy", "bellydrum", "conversion2", "haze", "healbell", "mist", "psychup", "refresh", "spite", "stockpile", "teleport", "transform"];
-				for (let k = 0; k < recoveryMoves.length; k++) {
-					let invalid = validParameter("moves", recoveryMoves[k], isNotSearch, target);
+				for (const move of recoveryMoves) {
```
Comment (Zarel) [inline chat-plugins/datasearch.js]: `moveid`

File: chat-plugins/trivia.js [hunk 2]
```diff
@@ -396,7 +396,7 @@ class Trivia extends Rooms.RoomGame {
 		// Trivia#destroy has already set the instance's room to null.
 		let tarRoom = this.room;
 		if (!tarRoom) {
-			for (let [roomid, room] of Rooms.rooms) { // eslint-disable-line no-unused-vars
+			for (const room of Rooms.rooms) { // eslint-disable-line no-unused-vars
```
Comment (Kaiepi) [inline chat-plugins/trivia.js]: `const [roomid room] of Rooms.rooms)1`

Comment (Morfent) [inline chat-plugins/trivia.js]: ` const [roomid, room] of Rooms.rooms)`

File: chat-plugins/mafia.js [hunk 3]
```diff
@@ -686,12 +685,12 @@ class Mafia extends Rooms.RoomGame {
 						} else {
```
Comment (Zarel) [inline chat-plugins/mafia.js]: um, three more lines up, what is that for-loop trying to do? it looks like a bug

Comment (Zarel) [reply chat-plugins/mafia.js]: Okay I thought more about what's going on. You should define `otherVoters = voters.filter(otherVoter => otherVoter !== voter);`.

That'll make the rest of this code extremely simple.

File: chat-plugins/trivia.js [hunk 4]
```diff
@@ -1580,8 +1577,8 @@ const commands = {
 
 		let questions = triviaData.questions;
 		let questionID = toId(question);
-		for (let i = 0; i < questions.length; i++) {
-			if (toId(questions[i].question) === questionID) {
+		for (const [i, question] of questions.entries()) {
```
Comment (Morfent) [inline chat-plugins/trivia.js]: `questtion.entries` isn't an array of `[i, question]`

File: chat-plugins/trivia.js [hunk 5]
```diff
@@ -1635,24 +1631,22 @@ const commands = {
...
 			return this.sendReply(buffer);
 		}
 
 		if (user.can('ban', null, room)) {
-			buffer += `<tr><td colspan="3">There are <strong>${listLen}</strong> questions in the ${CATEGORIES[target]} category for ${isWL ? "Weakest Link" : "Trivia"}.</td></tr>` +
+			buffer += `<tr><td colspan="3">There are <strong>${list.length}</strong> questions in the ${CATEGORIES[target]} category for ${isWL ? "Weakest Link" : "Trivia"}.</td></tr>` +
 				"<tr><th>#</th><th>Question</th><th>Answer(s)</th></tr>";
-			for (let i = 0; i < listLen; i++) {
-				let entry = list[i];
+			for (const [i, entry] of list.entries()) {
```
Comment (Morfent) [inline chat-plugins/trivia.js]: Ditto

File: chat-plugins/mafia.js (hunk 3 above)
Comment (Zarel) [reply chat-plugins/mafia.js]: Also OXFORD COMMA, WE'RE NOT SAVAGES

File: chat-plugins/modlog.js [hunk 6]
```diff
@@ -156,15 +156,15 @@ async function runModlog(rooms, searchString, exactSearch, maxLines) {
 	const useRipgrep = checkRipgrepAvailability();
 	let fileNameList = [];
 	let checkAllRooms = false;
-	for (let i = 0; i < rooms.length; i++) {
-		if (rooms[i] === 'all') {
+	for (const room of rooms) {
```
Comment (Zarel) [inline chat-plugins/modlog.js]: `roomid`

File: chat-plugins/modlog.js [hunk 7]
```diff
@@ -217,9 +217,8 @@ function runRipgrepModlog(paths, regexString, results) {
 	} catch (error) {
 		return results;
 	}
-	const fileResults = stdout.toString().split('\n').reverse();
-	for (let i = 0; i < fileResults.length; i++) {
-		if (fileResults[i]) results.tryInsert(fileResults[i]);
+	for (const file of stdout.toString().split('\n').reverse()) {
```
Comment (Zarel) [inline chat-plugins/modlog.js]: `fileName`

File: chat-plugins/trivia.js [hunk 8]
```diff
@@ -1635,24 +1631,22 @@ const commands = {
...
 				buffer += `<tr><td><strong>${(i + 1)}</strong></td><td>${entry.question}</td><td>${entry.answers.join(", ")}</td><tr>`;
 			}
 		} else {
-			buffer += `<td colspan="2">There are <strong>${listLen}</strong> questions in the ${target} category.</td></tr>` +
+			buffer += `<td colspan="2">There are <strong>${list.length}</strong> questions in the ${target} category.</td></tr>` +
 				"<tr><th>#</th><th>Question</th></tr>";
-			for (let i = 0; i < listLen; i++) {
-				buffer += `<tr><td><strong>${(i + 1)}</strong></td><td>${list[i].question}</td></tr>`;
+			for (const [i, entry] of list.entries()) {
```
Comment (Morfent) [inline chat-plugins/trivia.js]: Ditto

File: chat-plugins/trivia.js (hunk 2 above)
Comment (Zarel) [inline chat-plugins/trivia.js]: ...wat

Comment (Zarel) [reply chat-plugins/trivia.js]: Did you mean `Rooms.rooms.values()`?

Also you can remove the `eslint-disable-line`

File: chat-plugins/trivia.js [hunk 9]
```diff
@@ -1498,8 +1496,7 @@ const commands = {
 
 		if (toId(target) === 'all') {
 			if (isAccepting) {
-				for (let i = 0; i < submissionsLen; i++) {
```
Comment (Zarel) [inline chat-plugins/trivia.js]: Do we still need to define `submissionsLen`?

File: chat-plugins/trivia.js [hunk 10]
```diff
@@ -1609,14 +1606,13 @@ const commands = {
 			let questionsLen = questions.length;
 			if (!questionsLen) return this.sendReplyBox(`No questions have been submitted for ${isWL ? "Weakest Link" : "Trivia"} yet.`);
 
-			let categories = Object.keys(CATEGORIES);
 			let lastCategoryIdx = 0;
 			buffer += "<tr><th>Category</th><th>Question Count</th></tr>";
-			for (let i = 0; i < categories.length; i++) {
-				if (categories[i] === 'random') continue;
-				let tally = findEndOfCategory(categories[i], false) - lastCategoryIdx;
+			for (const category of Object.keys(CATEGORIES)) {
```
Comment (Zarel) [inline chat-plugins/trivia.js]: Honestly, `const category in CATEGORIES`...

`Object.keys` is superfluous if you're just going to iterate.

File: chat-plugins/trivia.js (hunk 4 above)
Comment (Zarel) [reply chat-plugins/trivia.js]: Assuming it's an array: yes it is? You know what `array.entries()` does, don't you?

File: chat-plugins/trivia.js (hunk 8 above)
Comment (Zarel) [reply chat-plugins/trivia.js]: Overruled

File: chat-plugins/trivia.js (hunk 5 above)
Comment (Zarel) [reply chat-plugins/trivia.js]: Overruled

File: chat-plugins/trivia.js (hunk 4 above)
Comment (Morfent) [reply chat-plugins/trivia.js]: ...sorry I wan't reading it right

File: chat-plugins/mafia.js (hunk 3 above)
Comment (KrisOnline) [reply chat-plugins/mafia.js]: i waSNT THE ONE WHO WROTE THE CHAT PLUGIN

**Comment by KrisOnline** [comment]
ok did

File: chat-plugins/trivia.js (hunk 9 above)
Comment (KrisOnline) [reply chat-plugins/trivia.js]: i removed it

**Comment by KrisOnline** [comment]
wtf? i removed it?

File: chat-plugins/trivia.js [hunk 11]
```diff
@@ -1609,14 +1605,13 @@ const commands = {
 			let questionsLen = questions.length;
 			if (!questionsLen) return this.sendReplyBox(`No questions have been submitted for ${isWL ? "Weakest Link" : "Trivia"} yet.`);
 
-			let categories = Object.keys(CATEGORIES);
 			let lastCategoryIdx = 0;
 			buffer += "<tr><th>Category</th><th>Question Count</th></tr>";
-			for (let i = 0; i < categories.length; i++) {
-				if (categories[i] === 'random') continue;
-				let tally = findEndOfCategory(categories[i], false) - lastCategoryIdx;
+			for (const category of CATEGORIES) {
```
Comment (Zarel) [inline chat-plugins/trivia.js]: This is still wrong...

Comment (Zarel) [reply chat-plugins/trivia.js]: I told you like fifty times, `in CATEGORIES`

File: chat-plugins/scavengers.js [hunk 12]
```diff
@@ -574,13 +574,13 @@ class ScavengerHunt extends Rooms.RoomGame {
 		if (questionArray.length % 2 === 1) return {err: "Your final question is missing an answer"};
 		if (questionArray.length < 6) return {err: "You must have at least 3 hints and answers"};
 
-		for (let i = 0; i < questionArray.length; i++) {
+		for (let [i, question] of questionArray.entries()) {
 			if (i % 2) {
-				questionArray[i] = questionArray[i].split(';').map(p => p.trim());
```
Comment (urkerab) [inline chat-plugins/scavengers.js]: I don't know this code so I'm wildly guessing that it depended on mutating the original array.

Comment (Zarel) [reply chat-plugins/scavengers.js]: Yep, it did!

**Pull request merged**